### ISSUE TITLE
Allow looking all the way up and down?

### ISSUE THREAD

**Issue by PopeRigby** [coelckers/Raze#416]
Would this be possible? Right now you can only look up and down about 45 degrees.

**Comment by mjr4077au** [comment]
~58 degrees for Duke and SW. It can be done with the new renderer but the weapon and hitscan code isn't up to it. There's a lot of work required to achieve this.

**Comment by coelckers** [comment]
Yeah, it's mostly due to poor coding on the game side. in GZDoom it never came up because ZDoom had solved the game logic problems years before. And in Doom it was far more isolated because the required core changes were in two functions only. Here the relevant code is a lot more scattered and in many cases so extremely convoluted that changing it can become a monumental task.

**Comment by Talon1024** [comment]
IIRC someone said it would be impossible to do this with Duke3D because it would require changes to the CON code.

**Comment by sinisterseed** [comment]
It would sure be worth a shot at it, but even so, that's not happening anytime soon. The new render needs to be in a solid state first, the cool stuff will only come later.

And that being said, it would be weird to have full up/down view in only some of the games. Wouldn't be surprised if Duke won't be capable of it if it requires CON changes, that pretty seals its fate.

**Comment by madame-rachelle** [comment]
Not necessarily. The CON code can be patched up - but this won't help mods that are using the same (albeit modified) CON code.

**Comment by mjr4077au** [comment]
I don't see why any CON changes will be needed. There's already horizon to pitch code in the engine, we just keep using that to adjust incoming CON values to pitch. We're already doing this now to a degree.

**Comment by madame-rachelle** [comment]
I think they were talking more about the projectiles themselves, not the projection which is purely handled by the engine anyhow.

**Comment by madame-rachelle** [comment]
So I figured out something that might be a bit of a solution to the CON problem but I have no idea how easy this really would be to implement, this might be a crapshoot but I know if this problem existed in the GZDoom engine itself this would be feasible.

The idea is this: We put in a custom directive for a CON file that signifies it for modern Raze use only. This way, we can distinguish it from classic CON files. When this directive is active, it is assumed that the author of the CON file has already corrected the tangent projectile issue and converted their pitch to proper trigonometry.

Then - within Raze itself there could be a patched CON file that has this fix, either that or a delta of some sort that allows this patch to be applied live. Mods could be checked for requiring said patch too - making it possible to patch older versions of more popular mods. Newer versions of the mods could ship with two CON files which, if exists a CON file that has a special naming scheme to indicate that it is for Raze, it will refuse to load the other version. This would allow for mod authors to create mods that are compatible both with Raze and with other engines.

Again, I don't know how feasible it is, but it's an idea that could be considered.

**Comment by sinisterseed** [comment]
> I think they were talking more about the projectiles themselves, not the projection which is purely handled by the engine anyhow.

Yea pretty much. Was referring to the actual gameplay impact, of course purely looking up/down fully would be possible even now, but that will be just a gimmick if the projectiles break or the game craps itself in other ways.
